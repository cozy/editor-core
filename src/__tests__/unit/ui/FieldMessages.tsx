import React from 'react';
import FieldMessages from '../../../ui/ConfigPanel/FieldMessages';
import { IntlProvider } from 'react-intl-next';
import { mount } from 'enzyme';

function mountMessage({ description }: { description: string }) {
  return mount(
    <IntlProvider locale="en">
      <FieldMessages description={description} />
    </IntlProvider>,
  );
}

function findHTML(wrapper: any) {
  const div = wrapper.find('Description div');
  const children = div.children();
  if (children.length) {
    return children;
  }
  return div.text();
}

describe('Field Messages', () => {
  it(' should support a plaintext description', async () => {
    const description = 'my description';
    const wrapper = mountMessage({ description });

    expect(findHTML(wrapper)).toMatchSnapshot();
  });

  it(' should support bold, italic, strong, em and code tags', async () => {
    const description =
      '<b>my</b> <i>description</i> that has <code>code</code> and <strong>strong</strong> <em>emphasis</em>';
    const wrapper = mountMessage({ description });

    expect(findHTML(wrapper)).toMatchSnapshot();
  });

  it(' should drop unsafe/unspecified tags', async () => {
    const wrapper = mountMessage({
      description: `<div>my</div> <a href='ohoh'>description</i> that has <strong><script src='ohno.js'></script>bad code and <em>emphasis</em></strong> on malformed <uNkoWn>tags`,
    });

    expect(findHTML(wrapper)).toMatchSnapshot();
  });
});
