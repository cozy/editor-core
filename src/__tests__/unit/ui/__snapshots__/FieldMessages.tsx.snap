// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Messages  should drop unsafe/unspecified tags 1`] = `
<span
  key="1"
>
  my
   
  <span
    key="2"
  >
    description that has 
    <strong
      key="1"
    >
      bad code and 
      <em
        key="2"
      >
        emphasis
      </em>
    </strong>
     on malformed 
    tags
  </span>
</span>
`;

exports[`Field Messages  should support a plaintext description 1`] = `"my description"`;

exports[`Field Messages  should support bold, italic, strong, em and code tags 1`] = `
<span
  key="1"
>
  <b
    key="0"
  >
    my
  </b>
   
  <i
    key="2"
  >
    description
  </i>
   that has 
  <code
    key="4"
  >
    code
  </code>
   and 
  <strong
    key="6"
  >
    strong
  </strong>
   
  <em
    key="8"
  >
    emphasis
  </em>
</span>
`;
